### PR TITLE
Add ApplicationContextInitializedEvent to correspond to contextPrepared in SpringApplicationRunListener

### PR DESCRIPTION
**Pull request by ayudovin**
this [enhancement](https://github.com/spring-projects/spring-boot/issues/14469)

### CODE REVIEW

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/event/ApplicationContextInitializedEvent.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/event/ApplicationContextInitializedEvent.java]: This should be 2012-2018.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/event/ApplicationContextInitializedEvent.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ */
+
+package org.springframework.boot.context.event;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+
+/**
+ * Event published as when a {@link SpringApplication} is starting up and the
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/event/ApplicationContextInitializedEvent.java]: I think this typo is in some of the other events' javadoc, but the "as" in "Event published as when a" is not needed here.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/event/ApplicationContextInitializedEvent.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * {@link ApplicationContext} is prepared but before any bean definitions are loaded.
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/event/ApplicationContextInitializedEvent.java]: I think it'd be good to mention that application context initialisers will have been called, something like:

ApplicationContext is prepared and ApplicationContextInitializers have been called but before any bean definitions are loaded.

**Comment by ayudovin** [comment]
@wilkinsona, Thank for your comment, I've added changes by your suggestion.

**Comment by wilkinsona** [comment]
Thanks once again, @ayudovin. The proposed changes are now in master.

**Comment by ayudovin** [comment]
@wilkinsona, Thank you.

**Pull request closed without merging**